(feat) Improve no active visit empty state in admission workspace

denniskigen opened the pull request:
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work is based on designs, which are linked or shown either in the Jira ticket or the description below.
- [x] My work includes tests or is validated by existing tests.

## Summary

This PR improves the flow for admitting a patient who has no active visit, following the design discussed in the review thread (and converging on the approach from [O3-4115](https://openmrs.atlassian.net/browse/O3-4115)):

- The admission requests workspace now shows a proper empty state with an illustration, helper text, and an "Add patient to ward" action when there are no pending requests.
- In the "Add patient to ward" patient search, result cards are no longer clickable wholesale. Each card renders an **Admit** button, which is disabled for patients without an active visit. A **Start visit** button is shown alongside so the user can start one first.
- After starting a visit from a search card, the admission workspace opens automatically for that patient instead of returning the user to the search results.
- The admit patient workspace now has a standard footer button bar (matching the bed selection form) and shows the active visit's type, location, and start time.
- The start-visit-then-admit flow exercised a workspace2 lifecycle race that was fixed in openmrs/openmrs-esm-core#1807; the framework version this branch resolves (via main's dependency updates) includes that fix.

## Screenshots
 
https://github.com/user-attachments/assets/02185ec0-a47e-4a38-9be0-abf4c53ab52d

## Related Issue

## Other

A note on the "already admitted" guard in the admit patient workspace: the disabled Admit button and warning are the only effective protection against duplicate admissions in this flow, not a convenience duplicating a backend rule. The ward app admits by posting a plain core encounter (`POST /ws/rest/v1/encounter`) with the emrapi admission encounter type. emrapi's "Patient is already admitted" check lives in `AdtService.createAdtEncounterFor` (see [AdtAction.java](https://github.com/openmrs/openmrs-module-emrapi/blob/master/api/src/main/java/org/openmrs/module/emrapi/adt/AdtAction.java#L109-L113)), which this path never touches. Core's encounter save applies no admission-state validation. Verified against a local backend: posting a second admission encounter for an already-admitted patient succeeds with a 201.

Related: patients who are already admitted still show an enabled "Admit" button on the search result cards, since the cards only know about visit status. Clicking through lands on the workspace warning with the action disabled (or a Transfer action if they're admitted to a different ward). Making the cards admission-aware would need an inpatient-admission lookup per search result, so I've left that as a possible follow-up.

<img width="2700" height="1978" alt="CleanShot 2026-07-16 at 01 20 24@2x" src="https://github.com/user-attachments/assets/786023f9-088c-483e-8a2f-bb3628273f8b" />

[O3-4115]: https://openmrs.atlassian.net/browse/O3-4115?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

chibongho reviewed (commented):
Should we just converge on the design [here](https://openmrs.atlassian.net/browse/O3-4115) in the queues app? So that:
- not the entire patient card is clickable
- we only show an "admit" button if they have an active visit; if not, we show a "Start visit" button instead.

jayasanka-sack commented:
Hi @denniskigen just checking, is this PR still valid / something you’re planning to continue?

P.S. I just checked. the issue still exists in dev3

denniskigen committed:
feat: improve no active visit admission state

Show a start visit prompt when a patient without an active visit is selected for admission.

Add scoped workspace styles and test coverage for the no active visit admission workflow.

denniskigen committed:
Show admit and start visit buttons on ward admission patient search cards

Implements the design agreed in the PR review discussion: in the Add
patient to ward search, the patient card is no longer clickable. Patients
with an active visit get an enabled Admit button that opens the admission
workspace. Patients without one get a start visit button next to a
disabled Admit button, which enables in place once the visit is saved.

The patient search workspace gains an optional selectPatientButton prop
so the ward can configure the button without leaking it into the other
consumers of the search workspace (appointments, patient lists, queues).

denniskigen committed:
Pass the patient to the start visit button slot on search result cards

The banner rendered start-visit-button-slot2 without the patient in the
slot state, so the start visit button launched the visit form with an
undefined patient prop. The visit form and the extensions it hosts, like
the billing check-in form, dereference that patient, which surfaced as
requests to patient/undefined and post-save callback failures on billing
enabled instances.

denniskigen committed:
Give the admit patient workspace a footer and visit context

The workspace rendered its primary action as a ghost link floating in an
otherwise empty panel, with a back link of equal visual weight beneath it
and no information beyond the patient banner. The admit and back actions
now sit in the standard workspace footer button bar, matching the bed
selection form, and the body shows the active visit's type, location,
and start time alongside the existing admission warnings.

AdmitPatientButton accepts optional kind and size props so the footer can
render it as a primary action without changing its ghost styling in the
admission requests bench.

denniskigen committed:
Auto-open the admission workspace after starting a visit

After the visit form saves in the Add patient to ward flow, the admission
workspace now opens automatically for the same patient instead of
returning the user to the search results to click Admit. The callback
flows from the ward's workspace props through the patient search context
and banner into the visit form's onVisitStarted hook.

Launching a child workspace immediately after another closes previously
tripped a lifecycle race in workspace2 that could mount the closed form's
component with the new workspace's props. That race was fixed in
openmrs/openmrs-esm-core#1807, which is included in the framework build
this branch resolves after rebasing onto main. Verified locally against
that build: repeated start-visit-then-admit flows land on the admission
workspace with no stale component mount and no patient/undefined
requests.

denniskigen committed:
Merge branch 'main' into feat/improve-admission-requests-workspace-empty-state

denniskigen committed:
Merge branch 'main' into feat/improve-admission-requests-workspace-empty-state

ibacher commented on packages/esm-ward-app/src/ward-workspace/create-admission-encounter/create-admission-encounter.scss:
```diff
@@ -0,0 +1,61 @@
+.buttonSet button {
+  max-width: unset !important;
+  width: 50% !important;
+}
```
This feels like something we might be able to do with a CSS grid or flex box and not strictly hard-coding the width?

ibacher reviewed (commented):
A couple of very minor things. Also I'm consistently getting a failure when I try to start a visit from the patientSearchApp:

```
Cannot read properties of null (reading 'patientUuid')
TypeError: Cannot read properties of null (reading 'patientUuid')
    at http://localhost:8080/openmrs/spa/openmrs-esm-patient-notes-app-12.3.5-pre.11649/8369.js:1:42891
    at ap (http://localhost:8080/openmrs/spa/98b428008990f466.js:1:59030)
    at s (http://localhost:8080/openmrs/spa/98b428008990f466.js:1:120044)
    at oO (http://localhost:8080/openmrs/spa/98b428008990f466.js:1:99678)
    at oD (http://localhost:8080/openmrs/spa/98b428008990f466.js:1:99548)
    at oE (http://localhost:8080/openmrs/spa/98b428008990f466.js:1:96297)
    at ox (http://localhost:8080/openmrs/spa/98b428008990f466.js:1:94852)
    at E (http://localhost:8080/openmrs/spa/98b428008990f466.js:1:131077)
    at MessagePort.L (http://localhost:8080/openmrs/spa/98b428008990f466.js:1:131598)
```

denniskigen committed:
Use flex instead of hard-coded button widths in admission workspace footer

Carbon's .cds--btn-set .cds--btn rule caps footer buttons at 196px and
out-specifies a single CSS module class, which is why the previous rule
needed !important. Doubling the class wins on specificity instead, and
flex: 1 splits the row evenly without hard-coding a 50% width. With two
buttons the layout is unchanged; in the error state the lone back button
now fills the footer instead of stopping at half width.

Addresses review feedback from @ibacher.

denniskigen committed:
Merge branch 'main' into feat/improve-admission-requests-workspace-empty-state

denniskigen committed:
Merge branch 'main' into feat/improve-admission-requests-workspace-empty-state
